common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -533,7 +534,7 @@ def should_force_account_creation():
             # account corresponds to them yet, if any.
             if should_force_account_creation():
                 return dispatch_to_register()
-            return dispatch_to_login()
+            login_user(strategy.request)
```

Comment:
couldn't this just be `student.views.login_user(strategy.request)` and then you could remove the new import on line 82?